src/IECoreImage/ClientDisplayDriver.cpp

```diff
@@ -96,11 +107,33 @@ ClientDisplayDriver::ClientDisplayDriver( const Imath::Box2i &displayWindow, con
 	m_data->m_host = displayHostData->readable();
 	m_data->m_port = displayPortData->readable();
 
+#if BOOST_VERSION >= 106600
+	boost::asio::io_context io_context;
+	tcp::resolver resolver(io_context);
+	boost::system::error_code error;
+	auto endpoints = resolver.resolve(m_data->m_host, m_data->m_port, error);
+	
+	if (!error)
+	{
+	    error = boost::asio::error::host_not_found;
+   	 	for (auto it = endpoints.begin(); it != endpoints.end() && error; ++it)
+    	{
+        	m_data->m_socket.close();
+        	m_data->m_socket.connect(*it, error);
+    	}
+	}
+
+	if (error)
+	{
+    	throw Exception("Could not connect to remote display driver server: " + error.message());
+	}
+#else
```

Comment:
As mentioned above, I think we should be able to have one version of the code here, that works for both Boost versions.